trapper/pipelines/question_answering_pipeline.py

```diff
@@ -188,112 +188,105 @@ def __init__(
         self._data_adapter = data_adapter
         self._data_collator = data_collator
 
-    def __call__(self, *args, **kwargs):
-        """
-        Find the answer(s) corresponding to the questions(s) in given context(s).
-
-        Args:
-            args (dict or a list of dicts):
-                One or several dicts containing the question and context.
-            question (:obj:`str` or :obj:`List[str]`): One or several question(s)
-                (must be used in conjunction with the :obj:`context` argument).
-            context (:obj:`str` or :obj:`List[str]`):
-                One or several context(s) associated with the answer(s) (must be
-                 used in conjunction with the :obj:`question` argument).
-            topk (:obj:`int`, `optional`, defaults to 1):
-                The number of answers to return (will be chosen by order of
-                likelihood).
-            max_answer_len (:obj:`int`, `optional`, defaults to 15):
-                The maximum length of predicted answers (e.g., only answers
-                with a shorter length are considered).
-            max_seq_len (:obj:`int`, `optional`, defaults to 384):
-                The maximum length of the total sentence (context)
-                after tokenization. The context will be
-                split in several chunks (using :obj:`doc_stride`) if needed.
-            max_question_len (:obj:`int`, `optional`, defaults to 64):
-                The maximum length of the question after tokenization. It will be
-                truncated if needed.
-            handle_impossible_answer (:obj:`bool`, `optional`, defaults to :obj:`False`):
-                Whether or not we accept impossible as an answer.
-
-        Return:
-            A list of :obj:`dict`: or a list of :obj:`dict`: Each result
-            comes as a dictionary with the following keys:
-            **score** (:obj:`float`) -- The probability associated to the
-                answer.
-            **answer** (:obj:`SpanTuple`) -- The answer corresponding to the
-                question.
-        """
-        self._set_default_kwargs(kwargs)
-        self._validate_kwargs(kwargs)
+    def _sanitize_parameters(
```

Comment:
Is there a reason why you didn't only use **kwargs here instead of named parameters. E.g. as shown in this example: https://huggingface.co/docs/transformers/add_new_pipeline#how-to-add-a-pipeline-to-transformers

Reply:
Replicated style from transformers code ([See here](https://github.com/huggingface/transformers/blob/v4.17.0/src/transformers/pipelines/question_answering.py#L165))